projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html

```diff
@@ -135,13 +135,13 @@
         [class.igx-grid__td--pinned]="col.pinned"
         [attr.aria-describedby]="gridID + '_' + col.field | igxStringReplace:'.':'_'"
         class="igx-grid__td igx-grid__td--fw igx-grid__td--tree-cell"
-        [class.igx-grid__td--edited]="rowID | transactionState:col.field:grid.rowEditable:grid.transactions:grid.pipeTrigger:grid.crudService.cell"
+        [class.igx-grid__td--edited]="rowID | transactionState:col.field:grid.rowEditable:grid.transactions:grid.pipeTrigger:grid.crudService.cell:grid.crudService.row"
         [class.igx-grid__td--number]="col.dataType === 'number' && col.visibleIndex !== 0"
         [ngClass]="col.cellClasses | igxCellStyleClasses:rowData[col.field]:rowData:col.field:viewIndex:grid.pipeTrigger"
         [ngStyle]="col.cellStyles | igxCellStyles:rowData[col.field]:rowData:col.field:viewIndex:grid.pipeTrigger"
         [expanded]="expanded"
-        [editMode]="col.editable && crudService.isCellInEditByPos(index, col.index)"
-        [addMode]="col.editable && crudService.isInAddMode(index,col.index)"
+        [editMode]="col.editable && crudService.targetInEdit(index, col.index)"
+        [addMode]="col.editable && crudService.targetInEdit(index, col.index)"
```

Comment:
If this is the exact same check as in `editMode`.. is it actually needed at all? @mddragnev ?

Reply:
IMO addMode input may be omitted. It is used only in cell component when the template getter is called. This was implemented because long ago we wanted to have row adding without editing as we thought to expose additional grid input. Now, we cannot have adding without editing so this is unnecessary. @Aleksandyr Feel free to remove it and refactor cell template get method.